tests/integration/conftest.py

```diff
@@ -136,3 +138,61 @@ def spec_conditionalized_version(tmp_path):
     specfile_path = tmp_path / SPECFILE
     shutil.copyfile(SPEC_CONDITIONALIZED_VERSION / SPECFILE, specfile_path)
     return specfile_path
+
+
+@pytest.fixture(
+    params=[
+        "file_path",
+        "text_file",
+        "content_string",
+        "text_io_stream",
+        "binary_io_stream",
+        "binary_file",
+    ]
+)
+def specfile_factory(request):
+    """
+    Pytest fixture to create a Specfile instance with different input modes.
+
+    :return: A function that creates a Specfile instance.
+    """
+    mode = request.param
+
+    def _create_specfile(input_path, **kwargs):
+        sourcedir_input = kwargs.get("sourcedir")
+        if sourcedir_input:
+            kwargs["sourcedir"] = Path(sourcedir_input)
+        else:
+            if mode == "file_path":
+                kwargs["sourcedir"] = Path(input_path).parent
+            elif mode in ["text_file_handle", "binary_file_handle"]:
+                kwargs.setdefault("sourcedir", Path(input_path).parent)
+            else:
+                kwargs.setdefault("sourcedir", Path(input_path).parent)
+
+        if mode == "file_path":
+            return Specfile(path=input_path, **kwargs)
+        elif mode == "text_file":
+            f = open(input_path, "r+", encoding="utf-8", errors="surrogateescape")
+            return Specfile(file=f, **kwargs)
+        elif mode == "content_string":
+            with open(input_path, encoding="utf-8", errors="surrogateescape") as f:
+                content = f.read()
+            return Specfile(content=content, **kwargs)
+        elif mode == "text_io_stream":
+            with open(input_path, encoding="utf-8", errors="surrogateescape") as f:
+                content = f.read()
+            string_io = io.StringIO(content)
+            return Specfile(file=string_io, **kwargs)
+        elif mode == "binary_io_stream":
+            with open(input_path, "rb") as f:
+                content_bytes = f.read()
+            bytes_io = io.BytesIO(content_bytes)
+            return Specfile(file=bytes_io, **kwargs)
+        elif mode == "binary_file":
+            f = open(input_path, "rb+")
+            return Specfile(file=f, **kwargs)
+        else:
+            raise ValueError(f"Unknown mode: {mode}")
```

Comment:
Again, it would make more sense to me to sort the conditions as I mentioned above.